test/unittests/test_excel_to_json_lists.py

```diff
@@ -0,0 +1,46 @@
+"""unit tests for excel to JSON list"""
+import os
+import unittest
+import json
+import jsonpath_ng
+import jsonpath_ng.ext
+import pandas
+
+from knora.dsplib.utils import excel_to_json_lists as e2l
+
+
+class TestExcelToJSONList(unittest.TestCase):
+
+    def test_excel2jsonlist(self) -> None:
+        # check that the output file was created
+        excelfolder = "testdata/lists"
+        outfile = "testdata/tmp/lists_output.json"
+        e2l.list_excel2json(listname=None, excelfolder=excelfolder, outfile=outfile)
+        self.assertTrue(os.path.exists(outfile), f'The outfile {outfile} was not created')
+
+        # check that the output file has the same number of nodes than the Excel file has rows
+        with open(outfile) as f:
+            output_as_dict = json.load(f)
+        output_nodes_matches = jsonpath_ng.parse('$..name').find(output_as_dict)
+        input_df = pandas.read_excel("testdata/lists/description_en.xlsx", header=None, dtype='str')
+        self.assertTrue(
+            len(input_df.index) == len(output_nodes_matches) - 1,
+            f"The output JSON file doesn't have the same number of nodes than the Excel file has rows"
+        )
+
+        # check that the longest Excel row(s) were correctly translated to the deepest-nested node(s)
+        longest_rows_indices = input_df[input_df.count().index[-1]].notna()
```

Comment:
Are you sure you need the `.notna()`? According to the documentation (https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.count.html) `count()` already ignores NA elements.